### ISSUE TITLE
Error using Yarn modern (berry) and pnpFallbackMode: none

### ISSUE THREAD

**Issue by tcodes0** [vitest-dev/vitest#3841]
### Describe the bug

A bare error is throw when attempting to run tests. From debugging it it seems to be just a `{}`, which causes some code accessing `error.message.length` to break. The underlying problem might be an unlisted dependency.
```sh
⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯ Unhandled Error ⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯
TypeError: Cannot read properties of undefined (reading 'length')
 ❯ printErrorMessage .yarn/__virtual__/vitest-virtual-c017bd09c3/0/cache/vitest-npm-0.33.0-420f3c7c44-c1884b2a1a.zip/node_modules/vitest/dist/vendor-cli-api.bf4b62a8.js:12519:21
 ❯ printError .yarn/__virtual__/vitest-virtual-c017bd09c3/0/cache/vitest-npm-0.33.0-420f3c7c44-c1884b2a1a.zip/node_modules/vitest/dist/vendor-cli-api.bf4b62a8.js:12407:3
 ❯ Logger.printError .yarn/__virtual__/vitest-virtual-c017bd09c3/0/cache/vitest-npm-0.33.0-420f3c7c44-c1884b2a1a.zip/node_modules/vitest/dist/vendor-cli-api.bf4b62a8.js:12628:12
 ❯ .yarn/__virtual__/vitest-virtual-c017bd09c3/0/cache/vitest-npm-0.33.0-420f3c7c44-c1884b2a1a.zip/node_modules/vitest/dist/vendor-cli-api.bf4b62a8.js:12685:18
 ❯ Logger.printUnhandledErrors .yarn/__virtual__/vitest-virtual-c017bd09c3/0/cache/vitest-npm-0.33.0-420f3c7c44-c1884b2a1a.zip/node_modules/vitest/dist/vendor-cli-api.bf4b62a8.js:12684:30
 ❯ DefaultReporter.printErrorsSummary .yarn/__virtual__/vites
 ```
 
 I've tracked it down to setting `pnpFallbackMode: none` on yarn (it's true by default). From the [official yarn docs](https://yarnpkg.com/features/pnp#fallback-mode)

> Back when PnP was first implemented, the compatibility wasn't as good as it is now. 
> To help with the transition, we designed a fallback mechanism: if a package tries to access an unlisted dependency, it's still allowed to resolve it if the top-level package lists it as a dependency. 
> We allow this because there's no resolution ambiguity, as there's a single top-level package in any project. 
> Unfortunately, this may cause confusing behaviors depending on how your project is set up. 
> When that happens, PnP is always right, and the only reason it works when not in a workspace is due to some extra lax.

What I have concluded so far is that there might be some part of vitest with an implicit dependency, but that dependecy must be in the reproduction's package json, because it works in fallback mode.

The fix might be as simple as adding a dependency to a package.json somewhere. :)
 

### Reproduction

https://github.com/tcodes0/vitest-yarn-repro

### System Info

```shell
System:
    OS: Linux 6.4 Arch Linux
    CPU: (6) x64 Intel(R) Core(TM) i5-9600K CPU @ 3.70GHz
    Memory: 15.46 GB / 31.29 GB
    Container: Yes
    Shell: 5.1.16 - /bin/bash
  Binaries:
    Node: 20.4.0 - ~/.nvm/versions/node/v20.4.0/bin/node
    Yarn: 3.6.1 - ~/.nvm/versions/node/v20.4.0/bin/yarn
    npm: 9.7.2 - ~/.nvm/versions/node/v20.4.0/bin/npm
```

### Used Package Manager

yarn

### Validations

- [X] Follow our [Code of Conduct](https://github.com/vitest-dev/vitest/blob/main/CODE_OF_CONDUCT.md)
- [X] Read the [Contributing Guidelines](https://github.com/vitest-dev/vitest/blob/main/CONTRIBUTING.md).
- [X] Read the [docs](https://vitest.dev/guide/).
- [X] Check that there isn't [already an issue](https://github.com/vitest-dev/vitest/issues) that reports the same bug to avoid creating a duplicate.
- [X] Check that this is a concrete bug. For Q&A open a [GitHub Discussion](https://github.com/vitest-dev/vitest/discussions) or join our [Discord Chat Server](https://chat.vitest.dev).
- [X] The provided reproduction is a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

**Comment by sheremet-va** [comment]
This is fixed in https://github.com/vitest-dev/vitest/commit/a78e6bd0e0d8d3a1a8c632189121b4122438579e

The fix will be a part of 0.34.0

**Comment by tcodes0** [comment]
I understand but I don't think it fixes the underlying error. It might just expose the real error. If still an issue in 0.34.0 I'll open another issue. thanks
